docs/system_overview/data_objects.md

```diff
@@ -0,0 +1,343 @@
+
+Data objects and replicas - as with any iRODS entity - are defined in the iRODS Catalog. The information is stored in the `R_DATA_MAIN` table. Each row in `R_DATA_MAIN` describes a single replica of a data object. Some system metadata is identical for each replica, such as data ID, associated collection, and name -- these are defining features of the data object which maps to the replica. Other system metadata varies between replicas, such as host resource and replica number -- these are defining features of the individual replicas.
+
+The following are all of the columns representing the system metadata in the table as well as their meanings:
+
+ - `data_id`: Identifier unique to the data object that maps to this replica. This column will contain the same value for all replicas of a given data object.
+ - `coll_id`: Identifier unique to the collection associated with the data object that maps to this replica. This column will contain the same value for all replicas of a given data object. The identifiers are defined in `R_COLL_MAIN` which contains all of the information about collections in a given zone.
+ - `data_name`: Logical Name of the data object that maps to this replica. This column will contain the same value for all replicas of a given data object. This value will be unique for all data objects with an identical `coll_id`.
+ - `data_repl_num`: Identifier unique to this replica. This value does not have any special meaning in the context of the iRODS system other than being able to identify a replica independently of its host resource.
+ - `data_version`: This column is not touched by iRODS anymore. Its purpose is lost to history.
+ - `data_type_name`:
+ - `data_size`: Size of the replica in bytes. This may vary between replicas depending on the state of the data and may or may not reflect the state of the data in storage. `ifsck` can be used to detect inconsistencies.
+ - `resc_group_name`: This column is not touched by iRODS anymore. It refers to the legacy resource groups feature. This should contain the value `EMPTY_RESC_GROUP_NAME` in order to discourage legacy queries.
+ - `resc_name`: This column is not touched by iRODS anymore. This is the name of the storage resource which hosts this replica. This should contain the value `EMPTY_RESC_NAME` in order to discourage legacy queries.
+ - `data_path`: Physical path of the data in storage for this replica. This is a string which is determined by the storage resource plugin hosting the data. For example, the unixfilesystem resource will specify a slash-delimited fully qualified path to a file on disk in the vault.
+ - `data_owner_name`: The username of the owner of the data object that maps to this replica. This column will contain the same value for all replicas of a given data object. This value is independent from the Access Control List for the data object which allows for multiple owners to exist. The value of this column will never change, even if the user listed is removed.
```

Comment:
"is removed" -> "is removed from the ACL"

Reply:
Done